app/server/app/routes/data.js

```diff
@@ -1,11 +1,153 @@
-const { resolve } = require("node:path");
-const { readFile } = require("node:fs/promises");
-const express = require("express");
 const axios = require("axios");
+const express = require("express");
+const { resolve } = require("node:path");
+const Papa = require("papaparse");
+const pg = require("pg");
+const QueryStream = require("pg-query-stream");
+const Excel = require("exceljs");
+const { getActiveSchema } = require("../middleware");
+const { knex } = require("../utilities/database");
 const logger = require("../utilities/logger");
 const log = logger.logger;
-const { getActiveSchema } = require("../middleware");
-const Assessment = require("../models").Assessment;
+const {
+  StreamingService,
+  ExcelTransform,
+} = require("../utilities/streamingService");
+
+function tryParseJSON(value) {
+  try {
+    return {
+      isJSON: true,
+      value: JSON.parse(value),
+    };
+  } catch (e) {
+    return {
+      isJSON: false,
+      value,
+    };
+  }
+}
+
+function appendToWhere(query, paramName, paramValue) {
+  if (!paramValue) return;
+
+  const parsedParam = tryParseJSON(paramValue);
+
+  if (Array.isArray(parsedParam.value)) {
+    query.whereIn(paramName, parsedParam.value);
+  } else {
+    query.where(paramName, parsedParam.value);
+  }
+}
+
+function parseCriteria(query, profile, queryParams) {
+  switch (profile) {
+    case "assessments":
+      appendToWhere(query, "id", queryParams.id);
+      appendToWhere(query, "reporting_cycle", queryParams.reportingCycle);
+      appendToWhere(query, "assessment_unit_id", queryParams.assessmentUnitId);
+      appendToWhere(
+        query,
+        "assessment_unit_name",
+        queryParams.assessmentUnitName
+      );
+      appendToWhere(query, "organization_id", queryParams.organizationId);
+      appendToWhere(query, "organization_name", queryParams.organizationName);
+      appendToWhere(query, "organization_type", queryParams.organizationType);
+      appendToWhere(query, "overall_status", queryParams.overallStatus);
+      appendToWhere(query, "region", queryParams.region);
+      appendToWhere(query, "state", queryParams.state);
+      appendToWhere(query, "ir_category", queryParams.irCategory);
+      break;
+    case "profile_test":
+      appendToWhere(query, "id", queryParams.id);
+      appendToWhere(query, "assessment_name", queryParams.assessmentName);
+      break;
+    default:
+      break;
+  }
+}
+
+async function executeQuery(profile, req, res, next) {
```

Comment:
The `next` parameter could be left off:
```suggestion
async function executeQuery(profile, req, res) {
```